### PR TITLE
[CI:DOCS] Cirrus: Support using updated/latest NV/AV in PRs

### PR DESCRIPTION
**Pull request by cevich**
On occasion, developers need to run the latest or bleeding-edge netavark/aardvark-dns in the podman CI environment.  Enable this through use of magic strings in the PR title, but only if the PR is marked as a draft.  The intent being, when the PR is ready for review, the current CI VM package versions will be used.  Hopefully also reminding the PR author to remove the magic strings from the title.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by flouthoc — APPROVED** [review]
LGTM

**Comment by flouthoc** [comment]
@cevich The build script is failing though

File: contrib/cirrus/lib.sh
Comment (flouthoc) [inline contrib/cirrus/lib.sh:256]:
```suggestion
            showrun dnf install -y netavark aardvark-dns
```

**Comment by cevich** [comment]
> I think this is missing the point that we want to keep the newer version for all PRs after that.

It's not safe to update the CI VM images in a podman PR since the images are used for multiple repos (i.e. buildah).  So incorporating a new/updated version would need to go through the c/automation_images build process to do what you're saying.  I realize this is less than ideal, but I can't think of another way to do it safely - only for podman CI.  We would need dedicated podman-only CI VM images instead of shared ones.  Possible to do, but a PITA to maintain.

What may help eventually is, I've been playing with using renovate to help update our CI VM images.  Such that they could be rebuilt often, like once per week, and renovate would open a PR in each repo. to run CI.  It's conceivable the build process could be automated at some point as well, but I think this is all quite a ways down the road.  The renovate setup itself seems pretty straight-forward, like something I could complete with a weeks-work or so (assuming renovate is enabled universally).

Unless I'm misunderstanding still, and/or maybe you have an idea how both stability and easy/tested updates, can be achieved?

**Comment by cevich** [comment]
Note: The method used in this PR results in nearly every single one of about 50 tasks, updating netavark/aardvark.  This is ripe for causing flakes if done frequently or on a large-scale.

**Comment by flouthoc** [comment]
@cevich I agree with you and also with @Luap99 , that once the PR goes green inorder to merge we will need podman CI to start using new netavark/aardvark. To me going the automation image route looks fine since it will not magically break things.

Like for example as of now I think podman's CI should start using recently released netavark/aardvark do you think we should update automation images.


> What may help eventually is, I've been playing with using renovate to help update our CI VM images. Such that they could be rebuilt often, like once per week, and renovate would open a PR in each repo. to run CI. It's conceivable the build process could be automated at some point as well, but I think this is all quite a ways down the road. The renovate setup itself seems pretty straight-forward, like something I could complete with a weeks-work or so (assuming renovate is enabled universally).
> 
> Unless I'm misunderstanding still, and/or maybe you have an idea how both stability and easy/tested updates, can be achieved?

Making renovate  do it once a week SGTM as a long term fix.

**Comment by cevich** [comment]
Okay good, yes I see the issue both ways as well.  I also recognize that rebuilding the CI VM images can be a daunting task.  I'm doing what I can to address the painful areas, at the same time I appreciate that the whole team benefits from CI VM images that are kept current.

Rolling out renovate is definitely going to be a longer-term prospect, and teaching it to manage CI VM images will be a further, follow-on addition to that effort.  I'm making progress, it's just slow going with keeping up on all my other tasks.  All that said, I'm fine either way if we want to just close this PR, or merge it as a short-term stop-gap, or if some other third option occurs to somebody.

File: contrib/cirrus/lib.sh
Comment (edsantiago) [inline contrib/cirrus/lib.sh:261]: Didn't get a proper chance to review today, so am filing first-impressions only: this disturbs me. It will cause confusion down the line. I believe it should look more like:
```
if CIRRUS_CHANGE_TITLE looks like this-or-that
    if not CIRRUS_PR_DRAFT
        die "Magic AVNV string can only be used on DRAFT PRs"
    fi

    ....now it's OK to check
```
I also think the `UPDATE` and `LATEST` strings are going to cause even more confusion: I _think_ the remember-trick is "UPDATE means dnf-update, the other one means copr"?? There has to be a more intuitive way to do this but I'm giving up for the day. Maybe `CI:NVAV=COPR` and `CI:NVAV=DNF`? Anyhow, bye for now.

**Comment by cevich** [comment]
> Maybe CI:NVAV=COPR and CI:NVAV=DNF? 

That's an interesting suggestion actually, probably would be easier to remember.

**Comment by edsantiago** [comment]
For extra credit, if this is possible, it'd be nice to make the check be `CI:[AN]V[AN]V=`, for those of us who won't remember whether "A" or "N" should be first

File: contrib/cirrus/lib.sh
Comment (cevich) [inline contrib/cirrus/lib.sh:258]: @edsantiago I thought it important to extract the kind of magic being requested so helpful error messages could be displayed.  I'm not sure if I'm extracting it the best way, though this seems to work :confused:

Comment (cevich) [reply contrib/cirrus/lib.sh:258]: [Example, catching 'foobar' magic.](https://cirrus-ci.com/task/5238151308902400)

File: contrib/cirrus/CIModes.md
Comment (edsantiago) [inline contrib/cirrus/CIModes.md:105]: I think you mean "temporary", or "timely", or even "ad-hoc"?

File: contrib/cirrus/lib.sh
Comment (edsantiago) [reply contrib/cirrus/lib.sh:258]: Nice.

File: contrib/cirrus/lib.sh
Comment (edsantiago) [inline contrib/cirrus/lib.sh:260]: I still find the wording confusing between these two messages (this and line 263). Sure, they look different side by side, but someone reading the log without benefit of both lines may have a hard time. Maybe something like this?
```bash
    warn "Updating ... from **fedora updates repo**"
...
    warn "Updating ... from **COPR build of main git branch**"
```

File: contrib/cirrus/lib.sh
Comment (edsantiago) [inline contrib/cirrus/lib.sh:261]: One could move lines 261 and 265, which are identical, down to just after line 268. If one were neurotic about removing unnecessary duplication, and about being polite to future maintainers who will otherwise spend time trying to understand what's the difference between those two `dnf` commands...

**Review by edsantiago — COMMENTED** [review]
LGTM. Minor nits, and of course you need to change the title :)

**Comment by TomSweeneyRedHat** [comment]
LGTM
once the failing tests are happy and @edsantiago 's suggestions are considered.

**Comment by cevich** [comment]
> once the failing tests are happy

They're failing on purpose...I'm testing the magic-string conditionals :smiley: Sorry if that wasn't clear.

File: contrib/cirrus/lib.sh
Comment (cevich) [reply contrib/cirrus/lib.sh:260]: Actually this reminded me.  We're **not** only updating from `updates`, sometimes we pull from `updates-testing`.  Let me make that clear as well...

**Comment by cevich** [comment]
[Example test failure](https://cirrus-ci.com/task/5597245303685120) when PR is not set as a **draft**.

**Comment by cevich** [comment]
Ugh...[this is going to annoy people](https://cirrus-ci.com/task/6003168971063296?logs=setup#L129) - Cirrus takes the original draft-mode state, not the current state.  So:

1. Developer pushes new PR, adds magic string to title
2. Developer goes to lunch anticipating 1.5 hours of testing
3. Instead, tests fail w/in 5-minutes because it's not draft-mode
4. Developer-on-lunch notices, and quickly switches to draft-mode and presses re-run.
5. Test fail again because the PR *was not previously* in draft-mode.
6. Annoyed-developer makes small commit change, and re-pushes.
7. Finally the developer can really go to lunch as the tests run.

I'l add a note to the docs about this.  I don't think there's much else I can do about it :cry:

**Comment by cevich** [comment]
[Example of installing](https://cirrus-ci.com/task/5932815393488896?logs=setup#L199) from `main` using podman-next COPR.

**Comment by edsantiago** [comment]
Examples (success and failure) are looking good. Thanks for testing.

The repush thing is a known feature of Cirrus: it gets a snapshot as of job-start time. That's why we can't just say "add NO NEW TESTS NEEDED to the github description and try again".

**Comment by cevich** [comment]
okay, I think (finally) this is ready.  Rebasing and pushing one last (required) time.

**Comment by edsantiago** [comment]
/lgtm

Thanks again!

**Pull request merged**
